mlir/lib/Target/Cpp/TranslateToCpp.cpp

```diff
@@ -1290,7 +1293,15 @@ std::string CppEmitter::getSubscriptName(emitc::SubscriptOp op) {
   llvm::raw_string_ostream ss(out);
   ss << getOrCreateName(op.getValue());
   for (auto index : op.getIndices()) {
-    ss << "[" << getOrCreateName(index) << "]";
+    ss << "[";
+    if (auto constant = dyn_cast_if_present<ConstantOp>(index.getDefiningOp());
+        constant && !shouldUseConstantsAsVariables()) {
+      assert(llvm::succeeded(emitAttributeToArbitraryStream(
```

Comment:
`assert(x)` will not be evaluated when LLVM is configured with assertions off, which means that `emitAttributeToArbitraryStream` would not run. You will need to move `emitAttributeToArbitraryStream` out of the assert.
For the assert itself: Is it possible to write valid IR which makes `emitAttributeToArbitraryStream` fail? If not, the assert is fine for me.

Reply:
It is possible to write valid IR, but invalid C++ code, when it fails. One example is to write nothing at all. Depending on the situation it could even write valid C++ code (e.g. `int v[5] = {0, 1};` vs `int v[] = {0, 1};`)

Reply:
Not relevant anymore after code refactor.